src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -331,6 +343,18 @@ public void Dispose()
                 }
             }
 
+            public void Cancel()
+            {
+                lock (_syncObject)
+                {
+                    if(!_disposed)
+                    {
+                        _cancelled = true;
+                        ValueTask ignored = _connection.SendRstStreamAsync(_streamId, Http2ProtocolErrorCode.StreamClosed);
```

Comment:
I don't think StreamClosed is the correct error code to send. InternalError might be best; this indicates that the cause of the RST_STREAM is internal to the client.

Reply:
Good catch. Looking at the spec, there's a dedicated error code for this called CANCEL that I think is appropriate:

>If the client determines, for any reason, that it does not wish to receive the pushed response from the server or if the server takes too long to begin sending the promised response, the client can send a RST_STREAM frame, using either the CANCEL or REFUSED_STREAM code and referencing the pushed stream's identifier.